src/main/java/aquality/selenium/core/elements/DefaultElementStateProvider.java

```diff
@@ -86,4 +110,21 @@ public boolean waitForEnabled(Duration timeout) {
     public boolean waitForNotEnabled(Duration timeout) {
         return isElementInDesiredCondition(elementNotEnabled(), timeout);
     }
+
+    private boolean doAndLogWaitForState(BooleanSupplier waitingAction, String conditionKeyPart, Duration timeout)
```

Comment:
looks strange that logging behaviour depends on timeout value

Reply:
it does, we don't want to log "wait" if we don't wait (ZERO timeout).